lib/web_ui/lib/src/engine/host_node.dart

```diff
@@ -110,11 +110,12 @@ class ShadowDomHostNode implements HostNode {
   /// This also calls [applyGlobalCssRulesToSheet], with the [defaultFont]
   /// to be used as the default font definition.
   ShadowDomHostNode(DomElement root, String defaultFont)
-      : assert(
-          root.isConnected ?? true,
-          'The `root` of a ShadowDomHostNode must be connected to the Document object or a ShadowRoot.'
-        ) {
-    _shadow = root.attachShadow(<String, dynamic>{
+      : assert(root.isConnected ?? true,
+            'The `root` of a ShadowDomHostNode must be connected to the Document object or a ShadowRoot.') {
+    final DomElement element =
+        domDocument.createElement('flt-shadow-host-node');
```

Comment:
Yeah that's correct, having a direct sibling to the `#shadow-root` caused issues rendering issues with those siblings.  I noticed the creation of the `input` listeners on the active element weren't being attached correctly. 

This is likely due to the entire subtree within `flt-glass-pane` getting replaced whenever `attachShadow` gets called on it - which is [standard shadowDOM behavior](https://web.dev/shadowdom-v1/#terminology-light-dom-vs-shadow-dom).  

Two ways around that:
1. Contain the #shadow-root within its own node, which is what I did above.  When `attachShadow` gets called on that node, only that node's children will get replaced, which allows the text-editing nodes to render correctly, which allow the inputs to function correctly.
2. We can use `<slot>`s within our `#shadow-root` to render any of its siblings. However, it's unclear whether text inputs rendered via `<slot>`s will face the same Chrome password autofill issues, since the `<slot>`s are technically within the shadowDOM.

Reply:
Yeah, IIRC in the presence of a shadow root direct children of the node are treated quite differently from normal elements. There's an expectation that there will be a `<slot>` inside the shadow root to project direct children into, because the elements are laid out based on constraints provided to the slot. Without one I'm not even sure how the element would be rendered. So moving text-editing-host into a separate branch is probably safer.

Reply:
nit: Would `<flt-render-host>` be a better name? It seems we're moving all the non-rendering things out of it, and it's used purely for positioning, laying out, and painting, which are all rendering aspects. Also, the `-node` suffix doesn't add useful information.

Reply:
I think that makes sense.  I pushed up the naming change and moved the semantics tree out of the text-editing host